packages/components/src/scroll-lock/index.js

```diff
@@ -1,114 +1,71 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useRef, useEffect } from '@wordpress/element';
 
 /**
- * Creates a ScrollLock component bound to the specified document.
- *
- * This function creates a ScrollLock component for the specified document
- * and is exposed so we can create an isolated component for unit testing.
- *
- * @param {Object} args Keyword args.
- * @param {HTMLDocument} args.htmlDocument The document to lock the scroll for.
- * @param {string} args.className The name of the class used to lock scrolling.
- * @return {WPComponent} The bound ScrollLock component.
+ * @param {string} className
+ * @return {(locked: boolean) => void} Function allowing you to set whether scrolling is locked.
  */
-export function createScrollLockComponent( {
-	htmlDocument,
-	className = 'lockscroll',
-} = {} ) {
-	if ( ! htmlDocument && typeof document !== 'undefined' ) {
-		htmlDocument = document;
-	}
-
-	let lockCounter = 0;
-
-	/*
-	 * Setting `overflow: hidden` on html and body elements resets body scroll in iOS.
-	 * Save scroll top so we can restore it after locking scroll.
-	 *
-	 * NOTE: It would be cleaner and possibly safer to find a localized solution such
-	 * as preventing default on certain touchmove events.
-	 */
-	let previousScrollTop = 0;
+function useSetLocked( className ) {
+	/** @type {import('react').MutableRefObject<number | undefined>} */
+	const previousScrollTop = useRef();
 
 	/**
-	 * Locks and unlocks scroll depending on the boolean argument.
-	 *
-	 * @param {boolean} locked Whether or not scroll should be locked.
+	 * @param {boolean} locked
 	 */
 	function setLocked( locked ) {
-		const scrollingElement =
-			htmlDocument.scrollingElement || htmlDocument.body;
+		if ( typeof document === undefined ) {
+			// if in SSR context, don't do anything
+			return;
+		}
```

Comment:
Do we need this check? I can never remember what runs in SRR, but if `componentDidMount` runs then we'll need this right, otherwise `setLocked` will run and `document` will be undefined.

Reply:
Gotcha, thanks for confirming. I'll remove the extra check.